app/v1/controllers/validators/AmendPensionChargesRulesValidator.scala

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.validators
+
+import api.controllers.validators.RulesValidator
+import api.controllers.validators.resolvers.{ResolveParsedCountryCode, ResolveParsedNumber}
+import api.models.errors._
+import cats.data.Validated
+import cats.data.Validated._
+import cats.implicits._
+import v1.models.request.AmendPensionCharges.{AmendPensionChargesRequestData, OverseasSchemeProvider, PensionCharges, PensionSavingsTaxCharges}
+
+object AmendPensionChargesRulesValidator extends RulesValidator[AmendPensionChargesRequestData] {
+  private val resolveParsedNumber      = ResolveParsedNumber()
+  private val qropsRefRegex            = "^[Q]{1}[0-9]{6}$"
+  private val pensionSchemeTaxRefRegex = "^\\d{8}[R]{1}[a-zA-Z]{1}$"
```

Comment:
```suggestion
  private val qropsRefRegex            = "^[Q]{1}[0-9]{6}$".r
  private val pensionSchemeTaxRefRegex = "^\\d{8}[R]{1}[a-zA-Z]{1}$".r
```

(to convert from string to regex)